### PR TITLE
ML API (DSS 4.2)

### PR DESCRIPTION
**Pull request by cstenac**

### CODE REVIEW

File: dataikuapi/dss/ml.py [hunk 1]
```diff
@@ -0,0 +1,572 @@
...
+
+        If K-Fold cross-test was used, most metrics will have a "std" variant, which is the standard deviation
+        accross the K cross-tested folds. For example, "auc" will be accompanied with "aucstd"
+
+        :returns: a dict of performance metrics values
+        :rtype: dict
+        """
+        import copy
+        clean_snippet = copy.deepcopy(self.snippet)
+        for x in ["gridsearchData", "trainDate", "topImportance", "backendType", "userMeta", "sessionDate", "trainInfo", "fullModelId", "gridLength", "algorithm", "sessionId"]:
```
Comment (instanceofme) [inline dataikuapi/dss/ml.py]: This looks difficult to remember to maintain, but I don't see a simple way around it :(.

Probably want to remove `topCoef` and `fullJobId` as well.

File: dataikuapi/dss/ml.py [hunk 2]
```diff
@@ -0,0 +1,572 @@
...
+
+class DSSMLTask(object):
+    """A handle to interact with a MLTask for prediction or clustering in a DSS visual analysis"""
+    def __init__(self, client, project_key, analysis_id, mltask_id):
+        self.client = client
+        self.project_key = project_key
+        self.analysis_id = analysis_id
+        self.mltask_id = mltask_id
+
+    def wait_guess_complete(self):
```
Comment (instanceofme) [inline dataikuapi/dss/ml.py]: Consider adding a timeout & raise an error on timeout? (or return False)

File: dataikuapi/dss/ml.py [hunk 3]
```diff
@@ -0,0 +1,572 @@
...
+
+        :return: a dict
+        """
+        return self.client._perform_json(
+                "GET", "/projects/%s/models/lab/%s/%s/status" % (self.project_key, self.analysis_id, self.mltask_id))
+                
+
+    def get_settings(self):
+        """
+        Gets the settings of this ML Tasks
```
Comment (instanceofme) [inline dataikuapi/dss/ml.py]: typo: Task

File: dataikuapi/dss/ml.py [hunk 4]
```diff
@@ -0,0 +1,572 @@
...
+    def start_train(self):
+        """
+        Starts asynchronously a new train session for this ML Task.
+
+        This returns immediately, before train is complete. To wait for train to complete, use ``wait_train_complete()``
+        """
+        self.client._perform_empty(
+                "POST", "/projects/%s/models/lab/%s/%s/train" % (self.project_key, self.analysis_id, self.mltask_id))
+
+    def wait_train_complete(self):
```
Comment (instanceofme) [inline dataikuapi/dss/ml.py]: Timeout?

Comment (kevanescence) [reply dataikuapi/dss/ml.py]: Agree with it :)

**Pull request merged**